server/api/views.py

```diff
@@ -242,3 +244,52 @@ def get(self, request, *args, **kwargs):
             return HttpResponseForbidden()
         else:
             return Response({'message': 'Authorization was a success'})
+
+
+class SlackActionsCallback(APIView):
+    authentication_classes = ()
+    permission_classes = (AllowAny,)
+
+    def post(self, request,):
+        data = request.data
+        payload = json.loads(data['payload'])
+        actions = payload.get('actions')
+
+        attend_action, action = self.check_action_type(actions, 'attend_event')
+        if attend_action:
+            event_id = int(action['value'])
+            user_slack_id = payload.get('user').get('id')
+
+            try:
+                event = Event.objects.get(id=event_id)
+                event_start_date = parse(event.start_date)
+                event_tz = timezone(event.timezone)
+                if event_start_date.tzinfo is None:
+                    event_start_date = event_start_date.replace(tzinfo=event_tz)
+                today = datetime.now(event_tz)
+                if today < event_start_date:
+                    user = AndelaUserProfile.objects.get(slack_id=user_slack_id)
+
+                    user_attendance = Attend(
+                        user=user,
+                        status='attending',
+                        event=event
+                    )
+                    user_attendance.save()
+
+                    message = generate_simple_message(
+                        'You\'ve been successfully registered as an attendee for the event :tada:')
```

Comment:
I am thinking we could have a shorter block-quoted message with an active tone here: `> You\'ve successfully registered for the event :tada:`

Reply:
I think I like this also